snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -343,4 +358,20 @@ public SnowflakeTableMetadata loadTableMetadata(SnowflakeIdentifier tableIdentif
   public void close() {
     connectionPool.close();
   }
+
+  private void tryMapSnowflakeExceptionToIcebergException(
+      SnowflakeIdentifier.Type identifierType, SQLException ex) {
```

Comment:
If you take the whole SnowflakeIdentifier here instead you can add the identifier to the exception message and make the message more useful, something like `"Identifier not found: '%s', underlying exception: '%s'"`